### PR TITLE
#1456 - Removed SelectModifierInterface and added CollectionProcessor 

### LINKED ISSUE #1456: Update media gallery data provider to utilize collection processor and migrate current filters/joins
Story #1291 

In order for the filters to be extensible, we will have to implement the collection processor and migrate the current filter and joins.

**Implementation details**
- Add virtual types for the collection processors (filter, sorting, pagination, join)
- Add collection factory and collection processor to the `MediaGalleryUi/Model/Listing/DataProvider.php` and remove the SelectModifierInterface.
- Migrate the IsLicensed (`AdobeStockImageAdminUi/Model/FilterParametersProvider/IsLicensed.php`) to a custom join using join processor
- Migrate the Directory (`MediaGalleryUi/Model/FilterParametersProvider/Directory.php`) to a custom filter using filter processor
- Migrate the KeywordsFilter (`MediaGalleryUi/Ui/Component/DataProvider/KeywordFilter.php`) to a custom filter using filter processor

**Technical details**
https://github.com/magento/adobe-stock-integration/wiki/MediaGalleryUi-DataProvider

### PR DESCRIPTION
**Pull request by gabrieldagama**
### Description (*)
Removed SelectModifierInterface and added CollectionProcessor to the MediaGalleryUi DataProvider.

### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1456: Update media gallery data provider to utilize collection processor and migrate current filters/joins

### Manual testing scenarios (*)
1. Go to `Content` -> `Media Gallery`
2. Test filters, pagination and sorting.

### CODE REVIEW

**Comment by gabrieldagama** [comment]
@magento run all tests

File: AdobeStockImageAdminUi/etc/adminhtml/di.xml [hunk 1]
```diff
@@ -33,11 +33,11 @@
             </argument>
         </arguments>
     </type>
-    <type name="Magento\MediaGalleryUi\Model\SelectModifierComposite">
+    <virtualType name="Magento\MediaGalleryUi\Model\Api\SearchCriteria\CollectionProcessor\JoinProcessor" type="Magento\Framework\Api\SearchCriteria\CollectionProcessor\JoinProcessor">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/etc/adminhtml/di.xml]: This virtual type is already declared in MediaGalleryUi module, so I think there is no need to specify the type
```suggestion
    <virtualType name="Magento\MediaGalleryUi\Model\Api\SearchCriteria\CollectionProcessor\JoinProcessor">
```

File: MediaGalleryUi/Model/Listing/DataProvider.php [hunk 2]
```diff
@@ -73,7 +83,6 @@ public function getData(): array
         try {
             /** @var SearchResult $searchResult */
             $searchResult = $this->getSearchResult();
-            $this->selectModifier->apply($searchResult->getSelect(), $this->getSearchCriteria());
 
             return $this->searchResultToOutput($searchResult);
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/Listing/DataProvider.php]: Unnecessary variable can be eliminated
```suggestion
            return $this->searchResultToOutput($this->getSearchResult());
```

File: MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/Keyword.php [hunk 3]
```diff
@@ -67,11 +60,11 @@ private function getSelectByKeyword(string $value): Select
                 ['asset_keywords_table' => $this->connection->getTableName(self::TABLE_ASSET_KEYWORD)],
                 ['id']
             )->where('keyword = ?', $value)
-             ->joinInner(
-                 ['keywords_table' => $this->connection->getTableName(self::TABLE_KEYWORDS)],
-                 'keywords_table.keyword_id = asset_keywords_table.id',
-                 ['asset_id']
-             ),
+                ->joinInner(
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/Keyword.php]: let's keep the consistent formatting in this method (let's move `)` to the next line)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @gabrieldagama , please see my review comments

**Comment by gabrieldagama** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed

Rechecked the:
- filters
- sorting
- pagination

**Pull request merged**